### PR TITLE
Redesign cost reporter: token-focused with phase inference

### PR DESCRIPTION
**Pull request by RutgerDijk**
## Summary

Redesign the cost reporter to focus on actionable token usage visibility rather than approximate dollar estimates. Shows where tokens go — which skills, phases, models, and branches consume the most — using compact bar-chart output.

## Changes

- **Core rework:** Strip all dollar pricing from `lib/cost-reporter.js`, replace with combined token counts
- **Phase inference:** Two-tier activity classification — explicit skill/agent labels + sliding-window tool-pattern inference (implementation, debugging, review, interactive, planning)
- **Bar-chart output:** Replace markdown tables with compact plain-text bar charts sorted by token count
- **Cross-project:** `--all` flag discovers all Claude projects and aggregates with BY PROJECT section
- **Team visibility:** Session-end hook exports summary JSON to `docs/costs/reports/`; `--team` flag aggregates committed reports with BY CONTRIBUTOR section
- **Hook registration:** New `cost-summary-export` hook registered in `hooks.json` and `hook-runner.js`

## Test Plan

- [x] 41 unit tests for cost-reporter (token aggregation, phase inference, bar-chart, cross-project, team aggregation)
- [x] 4 unit tests for cost-summary-export hook
- [x] Zero-division edge case for empty reports
- [x] Team report compatibility with formatReport

## Linked Issue

Closes #16

---

*Created with Envoy*

### CODE REVIEW

File: docs/plans/2026-04-06-cost-reporter-redesign.md [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+
+Adds a **BY PROJECT** section at the top. Other sections aggregate across all projects.
+
+### Team Visibility (`--team`)
+
+**Export (hook):** On session end, `hooks/cost-summary-export.js` writes a minimal JSON summary to `docs/costs/reports/<date>-<username>.json` containing:
+- Activity breakdown (label → token count)
+- Model breakdown (model → token count)
+- Branch breakdown (branch → token count)
+- Session metadata (date, turns, total tokens)
```
[the remark this answers is not in the document]
Comment (RutgerDijk) [reply docs/plans/2026-04-06-cost-reporter-redesign.md]: Fixed in `b70171c`. Plan doc now matches the actual hook payload (single branch string, not a map). Also updated filename format to include timestamp.
